src/Hl7.Fhir.ElementModel/Types/Any.cs

```diff
@@ -0,0 +1,182 @@
+/* 
+ * Copyright (c) 2019, Firely (info@fire.ly) and contributors
+ * See the file CONTRIBUTORS for details.
+ * 
+ * This file is licensed under the BSD 3-Clause license
+ * available at https://raw.githubusercontent.com/FirelyTeam/fhir-net-api/master/LICENSE
+ */
+
+#nullable enable
+
+using Hl7.Fhir.Utility;
+using System;
+
+namespace Hl7.Fhir.ElementModel.Types
+{
+    public abstract class Any
+    {
+        /// <summary>
+        /// Returns the concrete subclass of Any that is used to represent the
+        /// type given in parmameter <paramref name="name"/>.
+        /// </summary>
+        /// <param name="name"></param>
+        /// <param name="result"></param>
+        /// <returns></returns>
+        public static bool TryGetSystemTypeByName(string name, out Type? result)
+        {
+            result = get();
+            return result != null;
+
+            Type? get() =>
+                name switch
+                {
+                    "Any" => typeof(Any),
+                    "Boolean" => typeof(Boolean),
+                    "Code" => typeof(Code),
+                    "Concept" => typeof(Concept),
+                    "Decimal" => typeof(Decimal),
+                    "Integer" => typeof(Integer),
+                    "Long" => typeof(Long),
+                    "Date" => typeof(Date),
+                    "DateTime" => typeof(DateTime),
+                    "Ratio" => typeof(Ratio),
+                    "Time" => typeof(Time),
+                    "Quantity" => typeof(Quantity),
+                    "String" => typeof(String),
+                    "Void" => typeof(void),
+                    _ => null,
+                };
+        }
+
+        public static object Parse(string value, Type primitiveType)
+        {
+            if (value is null) throw new ArgumentNullException(nameof(value));
+
+            return TryParse(value, primitiveType, out var result) ? result! :
+                throw new FormatException($"Input string '{value}' was not in a correct format for type '{primitiveType}'.");
+        }
+
+        public static bool TryParse(string value, Type primitiveType, out object? parsed)
+        {
+            if (value == null) throw new ArgumentNullException(nameof(value));
+            if (!typeof(Any).IsAssignableFrom(primitiveType)) throw new ArgumentException($"Must be a subclass of {nameof(Any)}.", nameof(primitiveType));
+
+            bool success;
```

Comment:
It is. I want to avoid people trying `TryParse("1.3", typeof(FhirClient), out parsed)`.   The second argument must be a `Type` that is a subclass of `Any`.